finance/daml/DA/Finance/Asset.daml

```diff
@@ -85,6 +85,31 @@ setQuantity : Decimal -> AssetDeposit -> AssetDeposit
 setQuantity quantity assetDeposit =
     assetDeposit with asset = assetDeposit.asset with quantity = quantity
 
+
+
+makeCounterPartyObserver : MasterAgreement -> ContractId AssetDeposit -> Update (ContractId AssetDeposit)
+makeCounterPartyObserver = addRemoveCounterPartyObserver Action_Add
+  
+removeCounterPartyObserver : MasterAgreement ->  ContractId AssetDeposit -> Update (ContractId AssetDeposit)
+removeCounterPartyObserver = addRemoveCounterPartyObserver Action_Remove
+
+addRemoveCounterPartyObserver : AddRemove -> MasterAgreement ->  ContractId AssetDeposit -> Update (ContractId AssetDeposit)
```

Comment:
As the first arg, I would have rather used a function and name it as modifyCounterPartyObserver (higher level function).

That way, you don't really need AddRemove either

Reply:
makeCounterPartyObserver = modifyCounterPartyObserver DA.Set.insert

Reply:
Yeah, I was thinking about that, but I thought I wanted to limit what you can do with the function, i.e. given that I don't know of any other function that would make sense, I thought it would be easier to read it.